lib/fetch.ts

```diff
@@ -1,22 +1,42 @@
 import { client, meta } from "./grpc.ts";
 import { GetAllResponse } from "snitch-protos/protos/sp_external.ts";
-import { PipelineInfo } from "snitch-protos/protos/sp_info.ts";
+import {
+  ClientInfo,
+  LiveInfo,
+  PipelineInfo,
+} from "snitch-protos/protos/sp_info.ts";
 import { Audience } from "snitch-protos/protos/sp_common.ts";
 import { FlowEdge, FlowNode, mapEdges, mapNodes } from "./nodeMapper.ts";
+import { audienceKey } from "./utils.ts";
 
 export type ServiceMapType = GetAllResponse & {
   pipes: PipelineInfo[];
+  liveAudiences: Map<string, ClientInfo[]>;
 };
 
 export type ConfigType = { [key: string]: string };
 export type PipelinesType = { [key: string]: PipelineInfo };
 
+export const mapLiveAudiences = (live: LiveInfo[]) => {
+  const liveAudiences = new Map<string, ClientInfo[]>();
+
+  for (const l of live) {
```

Comment:
Nested loop is gross...but it comes to use in nested arrays and we have to iterate over all of them to de-dupe and store by audience key for downstream access.